google/resource_google_folder_test.go

```diff
@@ -0,0 +1,163 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"os"
+	"testing"
+
+	resourceManagerV2Beta1 "google.golang.org/api/cloudresourcemanager/v2beta1"
+)
+
+func TestAccGoogleFolder_rename(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+
+	folderDisplayName := "tf-test-" + acctest.RandString(10)
+	newFolderDisplayName := "tf-test-renamed-" + acctest.RandString(10)
+	org := os.Getenv("GOOGLE_ORG")
+	parent := "organizations/" + org
+	folder := &resourceManagerV2Beta1.Folder{}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleFolderDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGoogleFolder_basis(folderDisplayName, parent),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleFolderExists("google_folder.folder1", folder),
+					testAccCheckGoogleFolderParent(folder, parent),
+					testAccCheckGoogleFolderDisplayName(folder, folderDisplayName),
+				),
+			},
+			resource.TestStep{
+				Config: testAccGoogleFolder_basis(newFolderDisplayName, parent),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleFolderExists("google_folder.folder1", folder),
+					testAccCheckGoogleFolderParent(folder, parent),
+					testAccCheckGoogleFolderDisplayName(folder, newFolderDisplayName),
+				)},
+		},
+	})
+}
+
+func TestAccGoogleFolder_move(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+
+	folder1DisplayName := "tf-test-" + acctest.RandString(10)
+	folder2DisplayName := "tf-test-" + acctest.RandString(10)
+	org := os.Getenv("GOOGLE_ORG")
+	parent := "organizations/" + org
+	folder := &resourceManagerV2Beta1.Folder{}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleFolderDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGoogleFolder_basis(folder1DisplayName, parent),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleFolderExists("google_folder.folder1", folder),
+					testAccCheckGoogleFolderParent(folder, parent),
+					testAccCheckGoogleFolderDisplayName(folder, folder1DisplayName),
+				),
+			},
+			resource.TestStep{
+				Config: testAccGoogleFolder_move(folder1DisplayName, folder2DisplayName, parent),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleFolderExists("google_folder.folder1", folder),
+					testAccCheckGoogleFolderDisplayName(folder, folder1DisplayName),
+					testAccCheckGoogleFolderExists("google_folder.folder2", folder),
```

Comment:
Can you use a separate variable here for folder 2? Overwriting folder with the contents of folder 2 and then making similar assertions got really hard to read for me

Reply:
Done